Programs/brlapi_client.c

```diff
@@ -1662,10 +1662,29 @@ static int getControllingTty(void)
     FILE *stream;
 
     if ((stream = fopen("/proc/self/stat", "r"))) {
-      int vt;
-      int ok = fscanf(stream, "%*d %*s %*c %*d %*d %*d %d", &tty) == 1;
+      char buf[256];
+      size_t read = fscanf(buf, sizeof(*buf), sizeof(buf) / sizeof(buf[0]), stream);
```

Comment:
you meant fread here?

Reply:
Since we'd just read from the file and not scan, better just replace with open/read/close

Reply:
(also, with nowadays' C standard, sizeof(char) is always 1)

Reply:
> you meant fread here?

(put another way: please *test* your changes...)

Reply:
(as in: at the very least check that building doesn't spit warnings)

Reply:
As for actual testing, you can make the function non-static, and call it as a library function, it will work independently of the rest of the library.

Reply:
Tested the function body separately.

Reply:
Do we also have some github actions or sandbox tests that would ensure correctness ?